pkg/endpoint/events.go

```diff
@@ -119,6 +120,61 @@ func (e *Endpoint) PolicyRevisionBumpEvent(rev uint64) {
 	}
 }
 
+/// EndpointNoTrackEvent contains all fields necessary to update the NOTRACK rules.
+type EndpointNoTrackEvent struct {
+	ep     *Endpoint
+	annoCB AnnotationsResolverCB
+}
+
+// Handle handles the NOTRACK rule update.
+func (ev *EndpointNoTrackEvent) Handle(res chan interface{}) {
+	var IP string
+
+	e := ev.ep
+	port, err := ev.annoCB(e.K8sNamespace, e.K8sPodName)
+	if err != nil {
+		res <- &EndpointRegenerationResult{
+			err: err,
+		}
+		return
+	}
+
+	if !option.Config.EnableIPv4 {
+		IP = e.IPv6.String()
+	} else {
+		IP = e.IPv4.String()
+	}
+
+	// If this endpoint is going away, remove all related NOTRACK rules.
+	if err := e.lockAlive(); err != nil {
+		if e.noTrackPort != "" {
+			iptables.RemoveNoTrackRules(IP, e.noTrackPort)
+			e.noTrackPort = ""
+		}
+		res <- &EndpointRegenerationResult{
+			err: nil,
+		}
+		return
+	}
+
+	defer func() {
+		e.unlock()
+	}()
+
+	if port != e.noTrackPort {
+		if e.noTrackPort != "" {
+			iptables.RemoveNoTrackRules(IP, e.noTrackPort)
+		}
+		iptables.InstallNoTrackRules(IP, port)
```

Comment:
This needs to go under the `else` block?

Reply:
We should always install new rules even if there weren't any previously installed rules.